lib/ftp_sync.rb

```diff
@@ -2,24 +2,23 @@
 require 'fileutils'
 
 module FtpSync
-  
+
   # compares remote with local filelist
   # if local file not exists or older than remote file, download remote file
   # return array with new files
   def self.sync(supplier)
     new_files = Array.new
-    
-    # change local dir to save bnn-files correctly
-    FileUtils.mkdir_p(supplier.bnn_path) unless File.exists?(supplier.bnn_path)
-    Dir.chdir(supplier.bnn_path)
-    
+
+    # change local dir to save files correctly
+    FileUtils.mkdir_p(supplier.ftp_path) unless File.exists?(supplier.ftp_path)
+    Dir.chdir(supplier.ftp_path)
+
     # connect to ftp-server
-    ftp = Net::FTP.new(supplier.bnn_host, supplier.bnn_user, supplier.bnn_password)
-    
+    ftp = Net::FTP.new(supplier.ftp_host, supplier.ftp_user, supplier.ftp_password)
+
     # loop over the remote filelist
     ftp.nlst.each do |filename|
-      # File is an bnn article list? All BNN Files have filenames beginning with PL
-      if filename.match(/^(\.\/)?PL/)
+      if filename.match(Regexp.new(supplier.ftp_regexp))
```

Comment:
I should point out that sharedlist admins can configure `supplier.ftp_regexpr` without validation. I assume this to be safe because sharedlist admins can be trusted.